### ISSUE TITLE
No offline check when performing various Flutter CLI steps

### ISSUE THREAD

**Issue by oravecz** [flutter/flutter#138147]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

Our attempts to onboard Flutter into a major bank conflicts with our regulatory control procedures. We cannot use any applications which will go out to the public Internet - our networks do not allow this. 

Installing Flutter via git repository is a non-starter.

We begin with the Flutter SDK packages [available here](https://docs.flutter.dev/release/archive), and set the developers environment up thus

* PUB_HOSTED_URL set to our internal Pub server
* FLUTTER_STORAGE_BASE_URL set to our internal Google mirror
* Remove the Git origin remote
* Disable telemetry
* Disable analytics

As we do this, the calls to disable telemetry and disable analytics have long network timeouts associated with them before the command concludes.

The runner does not accept --offline or --no-version-check

```shell
/bin/flutter -vv --no-version-check --disable-telemetry
[   +5 ms] executing: sw_vers -productName
[  +30 ms] Exit code 0 from: sw_vers -productName
[        ] macOS
[        ] executing: sw_vers -productVersion
[  +20 ms] Exit code 0 from: sw_vers -productVersion
[        ] 13.6.1
[        ] executing: sw_vers -buildVersion
[  +25 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 22G313
[        ] executing: uname -m
[  +15 ms] Exit code 0 from: uname -m
[        ] arm64
[  +18 ms] executing: sysctl hw.optional.arm64
[  +21 ms] Exit code 0 from: sysctl hw.optional.arm64
[        ] hw.optional.arm64: 1
[  +37 ms] Analytics reporting disabled.
[+75107 ms] Running 0 shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 0
```

### Proposal

> This is an attempt to bring another use case to #129143. That ticket was closed after a discussion, but there isn't a summary of the discussion and the reasoning for the "won't fix" resolution. Can a summary behind the decision be posted on that ticket?

Proposing a global configuration (e.g. `--hermetic`) that we can use to truly prevent Flutter from making network calls. 

Ideally, when `--hermetic` is configured, the following actions take place

* Running `flutter upgrade` or `flutter downgrade` an attempt to upgrade/downgrade will result in a message indicating that external network requests have been disabled (Basically, any Git-related action - including channel switch - supports this message)
* New Flutter version checks do not occur
* Apparently, the delay I am seeing in disabling analytics is caused by this? "If you opt out of analytics, an opt-out event will be sent, and then no further information will be sent by the Flutter tool." - I would obviously not want that to happen if I have globally configured a `--hermetic` flag
* Perhaps the message "Flutter assets will be downloaded from $FLUTTER_STORAGE_BASE_URL. Make sure you trust this source!" could be disabled when the `--hermetic` flag is set. Or maybe, it could be mentioned on the initial Flutter command, and then no longer. It frightens the developers. :)

**Comment by danagbemava-nc** [comment]
Hi @oravecz, this appears to be a duplicate of https://github.com/flutter/flutter/issues/22429. 

Please feel free to share your thoughts in the issue linked above. 

Closing this in favour of the issue linked above.

If you disagree, kindly leave a comment below and I will reopen this. 

Thank you

**Comment by oravecz** [comment]
That ticket discusses how all commands should support the `--offline` option. 

In my case, imagine you have thousands of developers working in an environment that cannot access Google's FLUTTER_STORAGE_BASE_URL location. This is the standard, default behavior for these thousands of developers.

I am asking that instead of having them specify --offline on every single command, they can use `flutter config --offline` instead.

This command is executed one time, and all future calls will basically operate as if --offline is added as an option implicitly.

**Comment by danagbemava-nc** [comment]
Re-opening based on the comment above for further insight from the team

**Comment by christopherfujino** [comment]
Hi @oravecz, thanks for filing this issue. I'm not opposed to your proposal, however realistically it would take a lot of time to do that refactor that I am not currently prioritizing.

I wonder if there might be lower hanging fruit changes we could make that would unblock your use case (such as an ADDITIONAL flag to disable analytics WITHOUT the one-time event) in the near term, before a full-scale refactor to allow a mode where the tool never sends network requests. What do you think?

**Comment by christopherfujino** [comment]
cc @eliasyishak
